diff --git a/dist/vite-plugin-astro-server/css.js b/dist/vite-plugin-astro-server/css.js
index 8465a6abb392f397776bfb46cc9c37f14652d9f1..de912f77bbe1ddcc8aa6cc8b4d08c8d2a69b8c3d 100644
--- a/dist/vite-plugin-astro-server/css.js
+++ b/dist/vite-plugin-astro-server/css.js
@@ -30,9 +30,13 @@ async function getStylesForURL(filePath, loader) {
           continue;
         }
       }
+
+      const url = importedModule.url.replace(/^\0/, "/@id/__x00__");
+      const id = importedModule.id?.replace(/^0/, "/@id/__x00__");
+
       importedStylesMap.set(importedModule.url, {
-        id: importedModule.id ?? importedModule.url,
-        url: importedModule.url,
+        id: id ?? url,
+        url,
         content: css
       });
     }
